ggshield/cmd/sca/scan.py

```diff
@@ -0,0 +1,166 @@
+    directory: Optional[Path],
+    **kwargs: Any,
+) -> Optional[DummyReturnModel]:
+    """
+    Scan a directory for SCA vulnerabilities.
+    """
+    if directory is None:
+        directory = Path().resolve()
+
+    # Adds client and required parameters to the context
+    update_context(ctx)
+
+    result = sca_scan_all(ctx, directory)
+    return result
+
+
+def update_context(ctx: click.Context) -> None:
+    """
+    Basic implementation, should be populated with required parameters later.
+    """
+    config: Config = ctx.obj["config"]
+    ctx.obj["client"] = create_client_from_config(config)
+
+    # Empty for now
+    ctx.obj["exclusion_regexes"] = set()
+
+
+def sca_scan_all(ctx: click.Context, directory: Path) -> Optional[DummyReturnModel]:
+    """
+    Scan an entire directory for SCA Vulnerabilities.
+
+    - List SCA related files with a first call to SCA compute files API
+    - Create a tar archive with the required files contents
+    - Launches the scan with a call to SCA public API
+    """
+    client = SCAClient(ctx.obj["client"])
+
+    sca_filepaths = get_sca_scan_all_filepaths(
+        directory=directory,
+        exclusion_regexes=ctx.obj["exclusion_regexes"],
+        verbose=ctx.obj["config"].verbose,
+        client=client,
+    )
+
+    # No file to scan
+    if sca_filepaths is None:
+        display_info("\nNo file to scan.")
+        return
+
+    # empty for now
+    scan_parameters = SCAScanParameters()
+
+    # Call to full scan API and get results
+    scan_result = client.sca_directory_scan_all(
+        directory,
+        sca_filepaths,
+        scan_parameters,
+        ScanContext(
+            command_path=ctx.command_path,
+            scan_mode=ScanMode.SCA_DIRECTORY,
+        ).get_http_headers(),
+    )
+
+    # TODO add SCAScanResult when merged
+    if not scan_result.success or not isinstance(scan_result, DummyReturnModel):
+        if scan_result.status_code == 401:
+            raise APIKeyCheckError(client.base_uri, "Invalid API key.")
+        display_error("\nError scanning.")
+        display_error(str(scan_result))
+        return None
+
+    return scan_result
+
+
+def get_sca_scan_all_filepaths(
+    directory: Path,
+    exclusion_regexes: Set[re.Pattern],
+    verbose: bool,
+    client: SCAClient,
+) -> Optional[List[str]]:
+    """
+    Retrieve SCA related files of a directory.
+    First get all filenames that are not in blacklisted directories, then calls SCA compute files
+    API to filter SCA related files.
+    """
+    all_filepaths = get_all_files_from_sca_paths(
+        path=directory,
+        exclusion_regexes=exclusion_regexes,
+        verbose=verbose,
+        # If the repository is a git repository, ignore untracked files
+        ignore_git=False,
+    )
+
+    # API Call to filter SCA files
+    response = client.compute_sca_files(touched_files=all_filepaths)
+
+    if not response.success or not isinstance(response, ComputeSCAFilesResult):
```

Comment:
In which case can we have `response.success == True` and `not isinstance(response, ComputeSCAFilesResult)`?

Reply:
It seems that `response.success is True` only is status_code is `200`, whereas we can have `isinstance(response, ComputeSCAFilesResult)` for every `ok` codes. So looks like one of the two conditions is included in the other, and we could avoid the test on `response.success`.

Reply:
So why do you keep it (here and on line 129)?

Reply:
I just wanted to be sure. Removed.